sdk/java/pulumi/src/main/java/com/pulumi/resources/ResourceTransformation.java

```diff
@@ -1,78 +1,138 @@
 
 import javax.annotation.ParametersAreNonnullByDefault;
-import java.util.Objects;
 import java.util.Optional;
 
+import static java.util.Objects.requireNonNull;
+
+/**
+ * The callback signature for the {@code transformations} resource option.
+ * @see #apply(Args)
+ */
 public interface ResourceTransformation {
 
     /**
-     * ResourceTransformation#apply is the callback signature for @see {@link ResourceOptions#getResourceTransformations()}.
-     * A transformation is passed the same set of inputs provided to the @see {@link Resource} constructor,
-     * and can optionally return back alternate values for the "properties" and/or "options" prior to the resource
+     * ResourceTransformation#apply is the callback signature for {@link ResourceOptions#getResourceTransformations()}.
+     * A transformation is passed the same set of inputs provided to the {@link Resource} constructor,
+     * and can optionally return alternate values for the "properties" and/or "options" prior to the resource
      * actually being created. The effect will be as though those "properties" and/or
-     * "options" were passed in place of the original call to the @see {@link Resource} constructor.
+     * "options" were passed in place of the original call to the {@link Resource} constructor.
      *
-     * @return The new values to use for the "args" and "options" of the @see {@link Resource}
+     * @return The new values to use for the "args" and "options" of the {@link Resource}
      * in place of the originally provided values.
-     * Returns @see {@link Optional#empty()} if the resource will not be transformed.
+     * Returns {@link Optional#empty()} if the resource will not be transformed.
      */
     Optional<ResourceTransformation.Result> apply(ResourceTransformation.Args args);
 
+    /**
+     * The argument bag passed to a {@link Resource} transformation.
+     * @see #apply(Args)
+     */
     @ParametersAreNonnullByDefault
     class Args {
         private final Resource resource;
         private final ResourceArgs args;
         private final ResourceOptions options;
 
         public Args(Resource resource, ResourceArgs args, ResourceOptions options) {
-            this.resource = Objects.requireNonNull(resource);
-            this.args = Objects.requireNonNull(args);
-            this.options = Objects.requireNonNull(options);
+            this.resource = requireNonNull(resource);
+            this.args = requireNonNull(args);
+            this.options = requireNonNull(options);
+        }
+
+        /**
+         * @return the {@link Resource} instance that is being transformed.
+         */
+        public Resource resource() {
+            return this.resource;
         }
 
         /**
-         * The Resource instance that is being transformed.
+         * @return the {@link Resource} instance that is being transformed.
+         * @deprecated use {@link #resource()}
          */
+        @Deprecated
```

Comment:
didn't wanted to remove this right away since I'm not sure if examples or providers use it (unlikely?)

Reply:
Removing is OK but needs to be highlighted in CHANGELOG_PENDING to start building the habits. There is no signifiant use yet.